caikit/core/data_model/base.py

```diff
@@ -236,28 +263,36 @@ def parse_proto_descriptor(mcs, cls):
         #   registry shared for all
         _DataBaseMetaClass.class_registry[cls.full_name] = cls
 
-        # Add properties that use the underlying backend
-        # also add fields that existed in old_fields
-        # for supporting oneofs
+        # Add properties that use the underlying backend. Also add fields that
+        # existed in old_fields for supporting oneofs
         # see https://github.com/caikit/caikit/pull/107 for details
         for field in set(cls.fields + tuple(old_fields)):
-            setattr(cls, field, mcs._make_property_getter(field))
+
+            # If the field is the name of a field within a oneof and it was not
+            # in the old fields, the data is held under the oneof's name if this
+            # is the set value for the oneof
+            if oneof_name := cls._fields_to_oneof.get(field):
+                setattr(cls, field, mcs._make_property_getter(field, oneof_name))
```

Comment:
This extra argument results in a special getter that will first check `which_oneof` before returning the value of the attribute held under the name of the oneof